### PR TITLE
Move quick pick into editor override service

### PR DESCRIPTION
**Pull request by lramos15**
This is part of the additional debt work to cleanup the editor service's overrides. This PR moves the picker logic into the editor override service. It also removes the dependency of the editor service on the editor override service. The next step I believe is for all consumers of the `overrideOpenEditor` registration instead registers with the override service. This will allow me to remove the ugly patches that currently exist in the override service.

### CODE REVIEW

File: src/vs/workbench/browser/parts/editor/editorGroupView.ts [hunk 1]
```diff
@@ -804,6 +807,13 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 		return this.model.getEditors(order, options);
 	}
 
+	findEditors(resource: URI, order: EditorsOrder, options?: { excludeSticky?: boolean }): EditorInput[] {
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorGroupView.ts:810]: Can we add a test for this new method?

File: src/vs/workbench/browser/parts/editor/editorGroupView.ts [hunk 2]
```diff
@@ -804,6 +807,13 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
...
+		const canonicalResource = this.uriIdentityService.asCanonicalUri(resource);
+		return this.getEditors(order, options).filter(editor => {
+			return editor.resource && isEqual(editor.resource, canonicalResource);
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorGroupView.ts:813]: There is a `excludeSticky` option not being used here, do we actually need it?

File: src/vs/workbench/services/editor/browser/editorService.ts [hunk 3]
```diff
@@ -514,6 +515,26 @@ export class EditorService extends Disposable implements EditorServiceImpl {
...
 
+	private registerDefaultOverride(): void {
+		this._register(this.editorOverrideService.registerContributionPoint(
+			'*',
+			priorityToRank(ContributedEditorPriority.builtin),
+			{
+				id: DEFAULT_EDITOR_ASSOCIATION.id,
+				label: DEFAULT_EDITOR_ASSOCIATION.displayName,
+				detail: DEFAULT_EDITOR_ASSOCIATION.providerDisplayName,
+				active: (currentEditor) => this.fileEditorInputFactory.isFileEditorInput(currentEditor) && isEqual(currentEditor.resource, this.activeEditor?.resource),
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorService.ts:526]: Do we need to account for `SideBySideEditor` too where we want to probe on the right hand side?

File: src/vs/workbench/services/editor/browser/editorService.ts [hunk 4]
```diff
@@ -514,6 +515,26 @@ export class EditorService extends Disposable implements EditorServiceImpl {
...
+				priority: ContributedEditorPriority.builtin,
+			},
+			{},
+			(resource, editorID, options, group) => {
+				return { editor: this.createEditorInput({ resource }) };
+			},
+			(diffEditor, editorId, options, group) => {
+				return { editor: diffEditor };
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorService.ts:534]: What about the `options`?

File: src/vs/workbench/services/editor/browser/editorService.ts [hunk 5]
```diff
@@ -514,6 +515,26 @@ export class EditorService extends Disposable implements EditorServiceImpl {
...
+			{
+				id: DEFAULT_EDITOR_ASSOCIATION.id,
+				label: DEFAULT_EDITOR_ASSOCIATION.displayName,
+				detail: DEFAULT_EDITOR_ASSOCIATION.providerDisplayName,
+				active: (currentEditor) => this.fileEditorInputFactory.isFileEditorInput(currentEditor) && isEqual(currentEditor.resource, this.activeEditor?.resource),
+				priority: ContributedEditorPriority.builtin,
+			},
+			{},
+			(resource, editorID, options, group) => {
+				return { editor: this.createEditorInput({ resource }) };
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorService.ts:531]: What about the `options`?

File: src/vs/workbench/services/editor/browser/editorService.ts [hunk 6]
```diff
@@ -514,6 +515,26 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 		return overrides;
 	}
 
+	private registerDefaultOverride(): void {
+		this._register(this.editorOverrideService.registerContributionPoint(
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorService.ts:519]: It is not entirely clear to me why this is needed, would be great to clarify in a comment

File: src/vs/workbench/browser/parts/editor/editorGroupView.ts [hunk 7]
```diff
@@ -804,6 +807,13 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 		return this.model.getEditors(order, options);
 	}
 
+	findEditors(resource: URI, order: EditorsOrder, options?: { excludeSticky?: boolean }): EditorInput[] {
+		const canonicalResource = this.uriIdentityService.asCanonicalUri(resource);
+		return this.getEditors(order, options).filter(editor => {
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/editorGroupView.ts:812]: Looking at the users, we seem to not require the `order` for this method?

File: src/vs/workbench/services/editor/browser/editorService.ts [hunk 8]
```diff
@@ -579,14 +600,24 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 
 			// Override handling: pick editor or open specific
 			if (resolvedOptions?.override === EditorOverride.PICK || typeof resolvedOptions?.override === 'string') {
-				return this.openEditorWith(resolvedOptions.override, resolvedEditor, resolvedOptions, resolvedGroup);
+				const resolvedInputWithOptions = await this.editorOverrideService.resolveEditorOverride(editor as IEditorInput, resolvedOptions, resolvedGroup);
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorService.ts:603]: Not a fan of explicit casts, why is this needed?

File: src/vs/workbench/services/editor/browser/editorService.ts [hunk 9]
```diff
@@ -579,14 +600,24 @@ export class EditorService extends Disposable implements EditorServiceImpl {
...
+				return resolvedGroup.openEditor(resolvedInputWithOptions.editor, resolvedInputWithOptions.options ?? resolvedOptions);
 			}
 
 			// Override handling: ask providers to override
 			if (resolvedOptions?.override !== EditorOverride.DISABLED) {
-				const override = this.doOverrideOpenEditor(resolvedEditor, resolvedOptions, resolvedGroup);
-				if (override) {
-					return override;
+				// This will get cleaned up soon, but since the override service no longer uses the override flow we must check that
+				const resolvedInputWithOptions = await this.editorOverrideService.resolveEditorOverride(editor as IEditorInput, resolvedOptions, resolvedGroup);
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorService.ts:613]: Not a fan of explicit casts, why is this needed?

File: src/vs/workbench/services/editor/browser/editorService.ts [hunk 10]
```diff
@@ -579,14 +600,24 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 
 			// Override handling: pick editor or open specific
 			if (resolvedOptions?.override === EditorOverride.PICK || typeof resolvedOptions?.override === 'string') {
-				return this.openEditorWith(resolvedOptions.override, resolvedEditor, resolvedOptions, resolvedGroup);
+				const resolvedInputWithOptions = await this.editorOverrideService.resolveEditorOverride(editor as IEditorInput, resolvedOptions, resolvedGroup);
+				if (!resolvedInputWithOptions) {
+					return;
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorService.ts:605]:
```suggestion
					return undefined; // no editor was picked or registered for the identifier
```

File: src/vs/workbench/services/editor/browser/editorService.ts [hunk 11]
```diff
@@ -579,14 +600,24 @@ export class EditorService extends Disposable implements EditorServiceImpl {
...
+				}
+				return resolvedGroup.openEditor(resolvedInputWithOptions.editor, resolvedInputWithOptions.options ?? resolvedOptions);
 			}
 
 			// Override handling: ask providers to override
 			if (resolvedOptions?.override !== EditorOverride.DISABLED) {
-				const override = this.doOverrideOpenEditor(resolvedEditor, resolvedOptions, resolvedGroup);
-				if (override) {
-					return override;
+				// This will get cleaned up soon, but since the override service no longer uses the override flow we must check that
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorService.ts:612]: There seems to be some overlap with the above code, but you seem to indicate that this is being cleaned up, maybe add a `TODO` to not forget.

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/services/editor/browser/editorService.ts (hunk 11 above)
Comment (lramos15) [reply src/vs/workbench/services/editor/browser/editorService.ts:612]: Yeah I could clean up this patch a bit, but most of this will go away once everyone adopts it.

File: src/vs/workbench/services/editor/browser/editorService.ts (hunk 9 above)
Comment (lramos15) [reply src/vs/workbench/services/editor/browser/editorService.ts:613]: I cast this because my function expects an `IEditorInput`, but there's a chance that it's an `IResourceEditorType`. In that case I cast it which might not be the best solution and I should instead try to clean up the typings? I only really care about the resource, or in the case of the `DiffEditorInput` both resources.

File: src/vs/workbench/services/editor/browser/editorService.ts (hunk 8 above)
Comment (lramos15) [reply src/vs/workbench/services/editor/browser/editorService.ts:603]: See other comment :)

File: src/vs/workbench/browser/parts/editor/editorGroupView.ts (hunk 7 above)
Comment (lramos15) [reply src/vs/workbench/browser/parts/editor/editorGroupView.ts:812]: What do you mean we don't require order?

File: src/vs/workbench/services/editor/browser/editorService.ts (hunk 5 above)
Comment (lramos15) [reply src/vs/workbench/services/editor/browser/editorService.ts:531]: If the options aren't passed back then we use the ones that were passed in which I assume is what we want here

File: src/vs/workbench/services/editor/browser/editorService.ts (hunk 4 above)
Comment (lramos15) [reply src/vs/workbench/services/editor/browser/editorService.ts:534]: Same as below

File: src/vs/workbench/browser/parts/editor/editorGroupView.ts (hunk 2 above)
Comment (lramos15) [reply src/vs/workbench/browser/parts/editor/editorGroupView.ts:813]: It gets passed into the `getEditors` function which then gets passed into the model which cares

File: src/vs/workbench/services/editor/browser/editorService.ts (hunk 3 above)
Comment (lramos15) [reply src/vs/workbench/services/editor/browser/editorService.ts:526]: The override service handles this, but I forgot to make sure we could also override the group so I've fixed that.

File: src/vs/workbench/services/editor/browser/editorService.ts (hunk 6 above)
Comment (lramos15) [reply src/vs/workbench/services/editor/browser/editorService.ts:522]: Comment added.

File: src/vs/workbench/browser/parts/editor/editorGroupView.ts (hunk 7 above)
Comment (bpasero) [reply src/vs/workbench/browser/parts/editor/editorGroupView.ts:812]: I was wondering if `EditorsOrder` needs to be passed into `findEditors` at all if everyone asks for the same order

File: src/vs/workbench/services/editor/browser/editorService.ts (hunk 9 above)
Comment (bpasero) [reply src/vs/workbench/services/editor/browser/editorService.ts]: Yeah suggest to then write code to correctly distill the resource from either of the types as needed.

File: src/vs/workbench/browser/parts/editor/editorGroupView.ts (hunk 7 above)
Comment (lramos15) [reply src/vs/workbench/browser/parts/editor/editorGroupView.ts:812]: Ah I guess not. I'll fix that, I just figured for verbosity you could choose if you really wanted to.

File: src/vs/workbench/browser/parts/editor/editorGroupView.ts (hunk 1 above)
Comment (lramos15) [reply src/vs/workbench/browser/parts/editor/editorGroupView.ts]: Test added
